src/borg/crypto/low_level.pyx

```diff
@@ -439,22 +431,9 @@ cdef class AES256_CTR_BLAKE2b(AES256_CTR_BASE):
     cdef mac_compute(self, const unsigned char *data1, int data1_len,
                      const unsigned char *data2, int data2_len,
                      unsigned char *mac_buf):
-        cdef blake2b_state state
-        cdef int rc
-        rc = blake2b_init(&state, self.mac_len)
-        if rc == -1:
-            raise Exception('blake2b_init() failed')
-        with nogil:
-            rc = blake2b_update(&state, self.mac_key, 128)
-            if rc != -1:
-                rc = blake2b_update(&state, data1, data1_len)
-                if rc != -1:
-                    rc = blake2b_update(&state, data2, data2_len)
-        if rc == -1:
-            raise Exception('blake2b_update() failed')
-        rc = blake2b_final(&state, mac_buf, self.mac_len)
-        if rc == -1:
-            raise Exception('blake2b_final() failed')
+        data = bytes(self.mac_key[:128]) + bytes(data1[:data1_len]) + bytes(data2[:data2_len])
+        mac = hashlib.blake2b(data, digest_size=self.mac_len).digest()
+        mac_buf[:self.mac_len] = mac[:self.mac_len]
```

Comment:
the tests are crashing, likely here. does somebody see what's wrong?

Reply:
e9c4085 fixes this. is this a cython bug or was i expecting too much from its magic? @scoder

Reply:
See https://github.com/cython/cython/issues/2544
It's a bug that is specific to `char*` pointers (which are common enough to make this annoying).

Reply:
ah, at least it is known. \o/

Reply:
so, is a loop and byte-wise assignment the best way for now (for a few bytes)?

Reply:
If it's literally just a few bytes, then you probably won't notice a difference, regardless of how you do it. At some point, `memcpy()` would certainly beat it, but it can have its own bit of overhead. Non-ancient x86 CPUs have their own instruction for this, though. Can't say under which the C compiler decides to use it.